### PR TITLE
Pc implement index page

### PR DESCRIPTION
**Pull request by pconrad**
# Overview

In this PR, we implement a List page for the `UCSBDates` table.

We also introduce two fundamental elements into the code base:
* `useBackend` for getting data from the backend (we've refactored the Users page on the admin menu to use this as well
* `react-toastify` for putting up user notifications
 
# Issues Addressed

This closes issue #65

### CODE REVIEW

File: frontend/src/tests/pages/AdminUsersPage.test.js [hunk 1]
```diff
@@ -5,20 +5,28 @@ import AdminUsersPage from "main/pages/AdminUsersPage";
...
-        var axiosMock = new AxiosMockAdapter(axios);
+    var axiosMock = new AxiosMockAdapter(axios);
+
+    const resetAxiosMock = () => {
+        axiosMock.reset();
+        axiosMock.resetHistory();
         axiosMock.onGet("/api/currentUser").reply(200, apiCurrentUserFixtures.userOnly);
         axiosMock.onGet("/api/systemInfo").reply(200, systemInfoFixtures.showingNeither);
-        axiosMock.onGet("/api/admin/users").reply(200, usersFixtures.threeUsers);
+    }
```
Comment (btk5h) [inline frontend/src/tests/pages/AdminUsersPage.test.js:22]: Can we replace this with `beforeEach` ([docs](https://jestjs.io/docs/setup-teardown)) so we don't have to explicitly call it in tests?

File: frontend/src/tests/pages/AdminUsersPage.test.js [hunk 2]
```diff
@@ -5,20 +5,28 @@ import AdminUsersPage from "main/pages/AdminUsersPage";
...
 import axios from "axios";
 import AxiosMockAdapter from "axios-mock-adapter";
 
-describe("AdminUsersPage tests",  () => {
-    const queryClient = new QueryClient();
-    test("renders without crashing on two users", async () => {
+describe("AdminUsersPage tests", () => {
 
-        var axiosMock = new AxiosMockAdapter(axios);
+    var axiosMock = new AxiosMockAdapter(axios);
```
Comment (btk5h) [inline frontend/src/tests/pages/AdminUsersPage.test.js:15]: Can we make this `const` to be consistent?

File: frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    var axiosMock = new AxiosMockAdapter(axios);
+
+    const testId = "UCSBDatesTable";
+
+    const resetAxiosMock = () => {
+        axiosMock.reset();
+        axiosMock.resetHistory();
+        axiosMock.onGet("/api/currentUser").reply(200, apiCurrentUserFixtures.userOnly);
+        axiosMock.onGet("/api/systemInfo").reply(200, systemInfoFixtures.showingNeither);
+    }
```
Comment (btk5h) [inline frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js:25]: Same as above

File: frontend/src/tests/utils/useBackend.test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { QueryClient, QueryClientProvider } from "react-query";
+import { renderHook } from '@testing-library/react-hooks'
+import mockConsole from "jest-mock-console";
+
+// import usersFixtures from "fixtures/usersFixtures";
```
Comment (btk5h) [inline frontend/src/tests/utils/useBackend.test.js:5]: Is this comment necessary?

**Review by btk5h — CHANGES_REQUESTED** [review]

File: frontend/src/tests/pages/AdminUsersPage.test.js (hunk 2 above)
Comment (pconrad) [reply frontend/src/tests/pages/AdminUsersPage.test.js:15]: yes.. done

File: frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js (hunk 3 above)
Comment (pconrad) [reply frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js]: done

File: frontend/src/tests/utils/useBackend.test.js (hunk 4 above)
Comment (pconrad) [reply frontend/src/tests/utils/useBackend.test.js:5]: no, removed.

File: frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import { apiCurrentUserFixtures } from "fixtures/currentUserFixtures";
+import { systemInfoFixtures } from "fixtures/systemInfoFixtures";
+import { ucsbDatesFixtures } from "fixtures/ucsbDatesFixtures";
+import axios from "axios";
+import AxiosMockAdapter from "axios-mock-adapter";
+import mockConsole from "jest-mock-console";
+
+describe("UCSBDatesIndexPage tests", () => {
+
+    var axiosMock = new AxiosMockAdapter(axios);
```
Comment (bzamora020) [inline frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js:16]: axiosMock is constant in file, frontend/src/tests/pages/AdminUsersPage.test.js. Maybe we should make it constant here as well

File: frontend/src/tests/utils/useBackend.test.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+                },
+            },
+        })
+        const wrapper = ({ children }) => (
+            <QueryClientProvider client={queryClient}>
+                {children}
+            </QueryClientProvider>
+        );
+
+        var axiosMock = new AxiosMockAdapter(axios);
```
Comment (bzamora020) [inline frontend/src/tests/utils/useBackend.test.js:31]: Maybe we can also make "axiosMock" constant as well

**Review by bzamora020 — CHANGES_REQUESTED** [review]
Besides the const variable changes. It looks good to me!

File: frontend/src/tests/utils/useBackend.test.js (hunk 6 above)
Comment (pconrad) [reply frontend/src/tests/utils/useBackend.test.js]: good catch. I did this as a global change in the code base

File: frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js (hunk 5 above)
Comment (pconrad) [reply frontend/src/tests/pages/UCSBDates/UCSBDatesIndexPage.test.js]: good catch. I did this as a global change in the code base

**Pull request merged**